pytest_mh/_private/plugin.py

```diff
@@ -288,6 +293,13 @@ def pytest_collection_modifyitems(self, config: pytest.Config, items: list[pytes
                 deselected.append(item)
                 continue
 
+            # If mh_ignore_preferred_topology is True. Skip the preferred topology entirely.
+            if not self.mh_ignore_preferred_topology:
+                if item.get_closest_marker(name="preferred_topology") is not None:
+                    if not self._run_preferred_topology(item, data):
+                        deselected.append(item)
```

Comment:
This call should be moved to `_can_run_test` where the decision for other cases is made.